Sources/QuaternionModule/Transformation.swift

```diff
@@ -0,0 +1,415 @@
+  ///   ```
+  /// - If `vector` is `.infinity` or `-.infinity`, the quaternion is `.infinity`:
+  ///   ```
+  ///   Quaternion(rotation: -.infinity) == .infinity
+  ///   ```
+  ///
+  /// See also:
+  /// -
+  /// - `.angle`
+  /// - `.axis`
+  /// - `.angleAxis`
+  /// - `.polar`
+  /// - `.rotationVector`
+  /// - `init(angle:axis:)`
+  /// - `init(length:angle:axis)`
+  ///
+  /// - Parameter vector: The rotation vector.
+  ///
+  /// [wiki]: https://en.wikipedia.org/wiki/Axis–angle_representation#Rotation_vector
+  @inlinable
+  public init(rotation vector: SIMD3<RealType>) {
+    let angle: RealType = .sqrt(vector.lengthSquared)
+    if !angle.isZero && angle.isFinite {
+      self = Quaternion(halfAngle: angle/2, unitAxis: vector/angle)
+    } else {
+      self = Quaternion(angle)
+    }
+  }
+
+  /// Creates a quaternion specified with [polar coordinates][wiki].
+  ///
+  /// This initializer reads given `length`, `phase` and `axis` values and
+  /// creates a quaternion of equal rotation properties and specified *length*
+  /// using the following equation:
+  ///
+  ///     Q = (cos(phase), axis * sin(phase)) * length
+  ///
+  /// Given `axis` gets normalized if it is not of unit length.
+  ///
+  /// Edge cases:
+  /// -
+  /// - Negative lengths are interpreted as reflecting the point through the origin, i.e.:
+  ///   ```
+  ///   Quaternion(length: -r, angle: θ, axis: axis) == Quaternion(length: -r, angle: θ, axis: axis)
+  ///   ```
+  /// - For any `θ` and any `axis`, even `.infinity` or `.nan`:
+  ///   ```
+  ///   Quaternion(length: .zero, angle: θ, axis: axis) == .zero
+  ///   ```
+  /// - For any `θ` and any `axis`, even `.infinity` or `.nan`:
+  ///   ```
+  ///   Quaternion(length: .infinity, angle: θ, axis: axis) == .infinity
+  ///   ```
+  /// - Otherwise, `θ` and `axis` must be finite, or a precondition failure occurs.
+  ///
+  /// See also:
+  /// -
+  /// - `.angle`
+  /// - `.axis`
+  /// - `.angleAxis`
+  /// - `.rotationVector`
+  /// - `.polar`
+  /// - `init(angle:axis)`
+  /// - `init(rotation:)`
+  ///
+  /// [wiki]: https://en.wikipedia.org/wiki/Polar_decomposition#Quaternion_polar_decomposition
+  @inlinable
+  public init(length: RealType, phase: RealType, axis: SIMD3<RealType>) {
+    let axisLength: RealType = .sqrt(axis.lengthSquared)
+    if phase.isFinite && axisLength.isNormal {
+      self = Quaternion(
+        halfAngle: phase,
+        unitAxis: axis/axisLength
+      ).multiplied(by: length)
+    } else {
+      precondition(
+        length.isZero || length.isInfinite,
+        "Either angle must be finite, or length must be zero or infinite."
+      )
+      self = Quaternion(length)
+    }
+  }
+
+  /// Rotates given vector by this quaternion.
+  ///
+  /// As quaternions can be used to represent three-dimensional rotations, it is
+  /// is possible to also rotate a three-dimensional vector by a quaternion. The
+  /// rotation of an arbitrary vector by a quaternion is known as an action.
```

Comment:
Let's spell out what the action is a little bit more clearly here. `qvq⁻¹` should appear somewhere in this description =)

Reply:
I updated the comments and wrote down the action more clearly.